ouroboros-consensus/test-consensus/Test/Dynamic/Ref/RealPBFT.hs

```diff
@@ -0,0 +1,239 @@
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | A reference simulator of the RealPBFT protocol under \"ideal
+-- circumstances\"
```

Comment:
I like this idea :+1:

Reply:
Great, because I'm planning on doubling down on it! :D

I think we should have two primary test schemas: one where we generate as-rich-as-possible-yet-boring `TestConfig`s where we can predict the entire net's outcome (not sure how to handle multi-leader slots here), and one where we generate unrestrained `TestConfig`s and merely check the behavior of each node and each edge in isolation.

Reply:
That's a very similar strategy to what I'm following elsewhere for the p2p control governor. I agree it makes a lot of sense.